server/spec/test_external_resources_file.rb

```diff
@@ -38,16 +38,31 @@ def test_constructor
     assert_equal @erf.resources_base_directory, @resources_dir, "set external resources directory name correctly"
   end
 
-  def test_list_resources_top
+  def test_list_resources_top_nil
     parsed_file_list = JSON.parse(@erf.get_resource nil)
     assert_equal 2,parsed_file_list.length,"list resources in top level directory"
   end
 
+  def test_list_resources_top_empty_string
+    parsed_file_list = JSON.parse(@erf.get_resource "")
+    assert_equal 2,parsed_file_list.length,"list resources in top level directory"
+  end
+
   def test_list_resources_images
     parsed_file_list = JSON.parse(@erf.get_resource "image")
     assert_equal 8,parsed_file_list.length,"list resources in images directory"
   end
 
+  def test_list_resources_images_nil_file_name
```

Comment:
It means that if no specific file is requested a list of files will be returned.  I'm conflicted about how much documentation to put in test files.